central/cluster/datastore/datastore_impl.go

```diff
@@ -545,6 +545,8 @@ func (ds *datastoreImpl) postRemoveCluster(ctx context.Context, cluster *storage
 	ds.removeK8SRoles(ctx, cluster)
 	ds.removeRoleBindings(ctx, cluster)
 
+	// TODO: Apparently, cluster cves are only cleaned up at next cve fetch cycle.
```

Comment:
This seems like a good example of when we should use a hard constraint with cascade?

Reply:
Unfortunately no. Unless we move away from patterns like copyFrom we cannot.

Reply:
Yeah, this is fine, but we should refactor and reconsider a bunch of the stuff related to copy/etc